<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>713</width>
    <height>528</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Flash’n’Chips</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout" columnstretch="0,1,0">
      <item row="1" column="2">
       <widget class="QPushButton" name="detectBtn">
        <property name="text">
         <string>Detect devices</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="flashBtn">
        <property name="text">
         <string>Flash firmware</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="firmwareSelector"/>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="portSelector"/>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="selectPlatformLabel">
        <property name="text">
         <string>1) Select platform:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="selectFirmwareLabel">
        <property name="text">
         <string>3) Select firmware:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="platformSelector">
        <property name="currentText">
         <string notr="true">ESP8266</string>
        </property>
        <item>
         <property name="text">
          <string notr="true">ESP8266</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">CC3200</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="selectPortLabel">
        <property name="text">
         <string>2) Select port:</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QProgressBar" name="progressBar">
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="statusMessage">
      <property name="text">
       <string/>
      </property>
      <property name="textFormat">
       <enum>Qt::RichText</enum>
      </property>
      <property name="openExternalLinks">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="terminalControlsLayout" stretch="0,0,0,0,0,0">
      <property name="topMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QLabel" name="talkToDeviceLabel">
        <property name="text">
         <string>4) Use the console to talk to the device!</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="connectBtn">
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="rebootBtn">
        <property name="text">
         <string>Reboot</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clearBtn">
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="uploadBtn">
        <property name="text">
         <string>Upload file</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QPlainTextEdit" name="terminal">
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
      </property>
      <property name="maximumBlockCount">
       <number>4096</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="terminalInput">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="topMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QLabel" name="logo">
        <property name="text">
         <string>&lt;a href=&quot;https://www.cesanta.com/smartjs&quot;&gt;&lt;img src=&quot;:/images/smartjs_logo_small.png&quot;&gt;&lt;/a&gt;</string>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>© Cesanta, 2015</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>&lt;a href=&quot;https://github.com/cesanta/smart.js&quot;&gt;Smart.js documentation&lt;/a&gt;</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>713</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuConfigure">
    <property name="title">
     <string>Connect</string>
    </property>
    <addaction name="actionConfigure_Wi_Fi"/>
    <addaction name="actionOpenDashboard"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="actionOpenWebsite"/>
    <addaction name="actionSend_feedback"/>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionUpload_a_file"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuConfigure"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionConfigure_Wi_Fi">
   <property name="text">
    <string>Connect to Wi-Fi</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutQtRole</enum>
   </property>
  </action>
  <action name="actionOpenWebsite">
   <property name="text">
    <string>Smart.js website</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionUpload_a_file">
   <property name="text">
    <string>Upload</string>
   </property>
  </action>
  <action name="actionOpenDashboard">
   <property name="text">
    <string>Open dashboard</string>
   </property>
  </action>
  <action name="actionSend_feedback">
   <property name="text">
    <string>Send feedback</string>
   </property>
   <property name="menuRole">
    <enum>QAction::ApplicationSpecificRole</enum>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>platformSelector</tabstop>
  <tabstop>portSelector</tabstop>
  <tabstop>firmwareSelector</tabstop>
  <tabstop>flashBtn</tabstop>
  <tabstop>connectBtn</tabstop>
  <tabstop>rebootBtn</tabstop>
  <tabstop>clearBtn</tabstop>
  <tabstop>terminalInput</tabstop>
  <tabstop>terminal</tabstop>
  <tabstop>detectBtn</tabstop>
 </tabstops>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>clearBtn</sender>
   <signal>clicked()</signal>
   <receiver>terminal</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>619</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>576</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
